fabric-recipe-api-v1/src/main/java/net/fabricmc/fabric/api/recipe/v1/VanillaRecipeBuilders.java

```diff
@@ -0,0 +1,395 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.recipe.v1;
+
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+import it.unimi.dsi.fastutil.chars.Char2ObjectMap;
+import it.unimi.dsi.fastutil.chars.Char2ObjectOpenHashMap;
+import it.unimi.dsi.fastutil.chars.CharArraySet;
+import it.unimi.dsi.fastutil.chars.CharSet;
+
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemConvertible;
+import net.minecraft.item.ItemStack;
+import net.minecraft.recipe.BlastingRecipe;
+import net.minecraft.recipe.CampfireCookingRecipe;
+import net.minecraft.recipe.Ingredient;
+import net.minecraft.recipe.ShapedRecipe;
+import net.minecraft.recipe.ShapelessRecipe;
+import net.minecraft.recipe.SmeltingRecipe;
+import net.minecraft.recipe.SmokingRecipe;
+import net.minecraft.recipe.StonecuttingRecipe;
+import net.minecraft.tag.Tag;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.collection.DefaultedList;
+
+/**
+ * Provides some recipe builders for Vanilla recipes.
+ *
+ * @version 1.0.0
+ * @since 1.0.0
+ */
+public final class VanillaRecipeBuilders {
```

Comment:
Why is this class even here... This just deviates from our original intention to allow easy addition of any recipe.

Reply:
I think this is a good idea, its not perfect but it resolves one of the biggest issues with the origial design.

Reply:
Hmm, how applicable is this for 1.17 now all data gen stuff are available?

Reply:
Data gen is a whole other story, I havent really looked into much. Its always been in the jar even pre 1.17. I believe quite a few mods already make use of it. If there is expansion to be done there thats another topic.

This PR covers the cases where "pre-compiled" recipes dont do much good.